### ISSUE TITLE
Unable to get embedded resource via view in a specific scenario

### ISSUE THREAD

**Issue by gabriel-sevecek** [begriffs/postgrest#1044]
[instrospection.json.txt](https://github.com/begriffs/postgrest/files/1609752/instrospection.json.txt)

[schema.sql.txt](https://github.com/begriffs/postgrest/files/1609751/schema.sql.txt)
I'm using PostgREST 0.4.3.0 on macosx 10.13.2 and docker postgresql container with version 10.1. I have two similar views referencing the same table, one with join:

```
postgres=# \d+ api.fixtures
                                         View "api.fixtures"
     Column     |           Type           | Collation | Nullable | Default | Storage  | Description
----------------+--------------------------+-----------+----------+---------+----------+-------------
 id             | integer                  |           |          |         | plain    |
 competition_id | integer                  |           |          |         | plain    |
 home_team_id   | integer                  |           |          |         | plain    |
 away_team_id   | integer                  |           |          |         | plain    |
 result_id      | integer                  |           |          |         | plain    |
 date_time      | timestamp with time zone |           |          |         | plain    |
 status         | text                     |           |          |         | extended |
 matchday       | integer                  |           |          |         | plain    |
View definition:
 SELECT fixtures.id,
    fixtures.competition_id,
    fixtures.home_team_id,
    fixtures.away_team_id,
    fixtures.result_id,
    fixtures.date_time,
    fixtures.status,
    fixtures.matchday
   FROM data.fixtures;


postgres=# \d+ api.current_fixtures
                                     View "api.current_fixtures"
     Column     |           Type           | Collation | Nullable | Default | Storage  | Description
----------------+--------------------------+-----------+----------+---------+----------+-------------
 id             | integer                  |           |          |         | plain    |
 competition_id | integer                  |           |          |         | plain    |
 home_team_id   | integer                  |           |          |         | plain    |
 away_team_id   | integer                  |           |          |         | plain    |
 result_id      | integer                  |           |          |         | plain    |
 date_time      | timestamp with time zone |           |          |         | plain    |
 status         | text                     |           |          |         | extended |
 matchday       | integer                  |           |          |         | plain    |
View definition:
 SELECT f.id,
    f.competition_id,
    f.home_team_id,
    f.away_team_id,
    f.result_id,
    f.date_time,
    f.status,
    f.matchday
   FROM data.fixtures f
     LEFT JOIN data.competitions c ON f.competition_id = c.id
  WHERE f.matchday = c.current_matchday;
```

Selecting any embedded resource works on just only one of them if they're both present in schema. If I drop the one which works and restart postgrest, the other one starts working.
So for example this works:
```
GET 'http://localhost:3000/current_fixtures?select=id,competition(id,caption)'
```

But this doesn't:
```
$ GET 'http://localhost:3000/fixtures?select=id,competition(id,caption)'
HTTP/1.1 400 Bad Request
Content-Encoding: gzip
Content-Type: application/json; charset=utf-8
Date: Sun, 07 Jan 2018 12:27:24 GMT
Server: postgrest/0.4.3.0
Transfer-Encoding: chunked

{
"message": "Could not find foreign keys between these entities, No relation found between fixtures and competition"
}
```

But after `drop view api.current_fixtures;` and restarting postgrest
`GET 'http://localhost:3000/fixtures?select=id,competition(id,caption)'` works as expected.

schema and output from instrospection.sql attached.

**Comment by steve-chavez** [comment]
This is the same problem reported in #823, #562. Basically the relation cannot be found due to a second view. I've found that the [raiseRelations](https://github.com/begriffs/postgrest/blob/master/src/PostgREST/DbStructure.hs#L282-L283) function is not adding a `Parent` relation in this case.

**Comment by chrstphrhrt** [comment]
Minor note, we ran into this error and took a while to debug. Turns out is was user error for not requesting the right RLS-scoped views in the URL. The response message was that relation doesn't exist so we went down the analysis rabbit-hole to make sure we didn't have conflicting view names or anything.

It would be great if RLS-scoped request errors wouldn't say 400 relation doesn't exist, but rather something like "403 permission denied for relation parent -> child".
